src/views/pages/AddUser/index.js

```diff
@@ -140,33 +141,73 @@ class AddUser extends Component {
-            .then(res => this.setState({ newUser: this.props.user, submitted: true }));
+            .then(res => this.setState({ newUser: this.props.user, submitted: true, createFailed: false }));
     };
 
     redirectToExplore = () => <Redirect to="/explore" />;
 
+    handleFailedCreate = () => {
+        const userStatus = _.get(this.props.user, 'status', NOT_STARTED);
+        if (userStatus === ERROR) {
+            if (!this.state.createFailed) {
+                this.refs.fullName.value = '';
+                this.refs.userName.value = '';
+                this.refs.password.value = '';
+
+                const defaultNewUser = {
+                    name: '',
+                    username: '',
+                    password: '',
+                    address: {
+                        street: '',
+                        city: '',
+                        state: '',
+                        zipcode: '',
+                        country: ''
+                    },
+                    contactInformation: {
+                        phoneNumber: '',
+                        email: ''
+                    },
+                    website: '',
+                    description: '',
+                    organizations: []
+                };
+
+                this.setState({ createFailed: true, newUser: defaultNewUser });
+            }
+        }
+    };
+
+    getErrorMessage = () => {
+        if (_.get(this.props.user, 'error.message', null) === null) {
+            return 'An error occurred during creation. Please contact and administrator.';
+        }
+        return this.props.user.error.message;
+    };
+
     renderForm = () => {
         return (
             <Card>
                 <div className="card-body">
                     <div className="form-group">
                         <label htmlFor="name">Full Name</label>
                         <input
-                            type="text" className="form-control" id="fullName"
+                            type="text" className="form-control" id="fullName" ref="fullName"
                             onChange={(e) => this.handleFormInput(e)}
                         />
                     </div>
                     <div className="form-group">
                         <label htmlFor="username">Username</label>
                         <input
-                            type="text" className="form-control" id="userName"
+                            type="text" className="form-control" id="userName" ref="userName"
                             onChange={(e) => this.handleFormInput(e)}
                         />
                     </div>
                     <div className="form-group">
                         <label htmlFor="password">Password</label>
                         <input
-                            type="text" className="form-control" id="password"
+                            type="text" className="form-control" id="password" ref="password"
```

Comment:
React 16 has a slightly different ref approach, this style of refs does not bind to a react instance and causes a bunch of problems. Here's how refs are used in this version of React: https://reactjs.org/docs/refs-and-the-dom.html

As a bonus, ID's in React are sort of a no-no. IT's fine because this component should be the only instance ever, but if you were to render two on the same DOM it would not be super happy. Just check if there's any references to those ID's and check styling and should be fine to remove.

Reply:
As a big giant thumbs up you could replace with Redux-Forms which is super cool but that's not at all expected 🙂

Reply:
I could give Redux-Forms a try next time! I'm just getting close to being done for today, so I'm not sure I want to dive in right now. I'll check out the ID situation.

Reply:
Im not at all concerned about it. It'll happen at some point. It's on my list of wants

Reply:
Hmm, looks like the function 'handleFormInput' uses the 'id' to determine how to handle updates to fields. May need some more work to figure out if ids can be taken out or not.

Reply:
Everywhere you put see references to the ID you can replace with `this.refs.myRef` but yeah